include/fastrtps/types/DynamicTypePtr.h

```diff
@@ -54,6 +54,16 @@ class DynamicType_ptr : public std::shared_ptr<DynamicType>
 
     RTPS_DllAPI DynamicType_ptr& operator =(
             DynamicType*);
+
+    RTPS_DllAPI bool operator!=(std::nullptr_t) const
+    {
+            return bool(*this);
+    }
+
+    RTPS_DllAPI bool operator==(std::nullptr_t) const
+    {
+            return !*this;
+    }
```

Comment:
Adding new API should add corresponding tests.

Reply:
Already checked in \test\unittest\dynamic_types\DynamicTypesTests.cpp